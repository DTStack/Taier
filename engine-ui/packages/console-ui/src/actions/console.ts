import API from '../api/account'
import { userActions } from '../consts/consoleActions'

// Action
export function getTenantList () {
    return (dispatch: any) => {
        API.getFullTenants().then((res: any) => {
            if (res.success) {
                return dispatch({
                    type: userActions.SET_TENANT_LIST,
                    data: res.data || []
                })
            }
        })
    }
}
