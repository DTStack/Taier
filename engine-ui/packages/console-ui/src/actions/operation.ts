import api from '../api/operation'
import { hashHistory } from 'react-router'

export const workbenchActions = (dispatch?: any) => {
    return {
        openTaskInDev: (data: any) => {
            // 项目类型
            api.getYWAppType({
                id: data.id
            }).then((res: any) => {
                if (res.code === 1) {
                    console.log('data: ', data);
                }
            });
            // let path2 = `http://dev.insight.dtstack.cn/batch/#/offline/task?taskId=${data.id}&pid=${data.projectId}`
            // window.open(path2)
            if (data.appType == '10') {
                if (data.taskType == 18) {
                    let path2 = `http://${window.location.hostname}:8099/easy-index/index-define?taskId=${data.id}`;
                    window.open(path2)
                } else if (data.taskType == 7) {
                    hashHistory.push({ pathname: '/operation/dependence', query: { id: data.id } });
                }
            }
        }
    }
}
