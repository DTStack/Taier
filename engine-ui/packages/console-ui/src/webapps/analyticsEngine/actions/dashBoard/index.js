import { message } from 'antd';
import API from '../../api/dashBoard';

