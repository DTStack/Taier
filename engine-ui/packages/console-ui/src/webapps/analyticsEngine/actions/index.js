import commonActionType from "../consts/commonActionType";
import API from "../api";

const commonActions = {
    getUserList(params) {
        return dispatch => {
            API.getUserList(params).then(res => {
                if (res.code === 1) {
                    dispatch({
                        type: commonActionType.GET_USER_LIST,
                        payload: res.data
                    });
                }
            });
        };
    },
    getAllDict(params) {
        return dispatch => {
            API.getAllDict(params).then(res => {
                if (res.code === 1) {
                    dispatch({
                        type: commonActionType.GET_ALL_DICT,
                        payload: res.data
                    });
                }
            });
        };
    }
};

export default commonActions;
