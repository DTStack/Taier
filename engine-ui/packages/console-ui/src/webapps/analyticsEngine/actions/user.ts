import UserApi from '../api/user'
import userActions from '../consts/userActions'

// Action
export function getUser () {
    return (dispatch: any) => {
        UserApi.test()
    }
}

export function updateUser (fields: any) {
    return {
        type: userActions.UPDATE_USER,
        data: fields
    }
}

export function getUserList (fields: any) {
    return (dispatch: any) => {
    }
}
