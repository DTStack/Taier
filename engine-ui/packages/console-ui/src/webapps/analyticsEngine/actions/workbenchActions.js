import API from '../api';
import moment from 'moment';
import workbenchAction from '../consts/workbenchActionType';
import modalAction from '../consts/modalActionType';
import { notification } from 'antd';

// import { CATALOGUE_} from '../consts';

export const updateModal = (value) => {
    return { type: modalAction.UPDATE_MODAL, data: value }
}

export const resetModal = () => {
    return { type: modalAction.RESET_MODAL }
}


export function switchTab(currentTab) {
    return {
        type: workbenchAction.SWITCH_TAB,
        payload: parseInt(currentTab, 10)
    }
}

export function openTab(tabData) {
    return {
        type: workbenchAction.OPEN_TAB,
        payload: tabData,
    }
}

export function closeTab(currentTab) {
    return {
        type: workbenchAction.CLOSE_TAB,
        payload: parseInt(currentTab, 10)
    }
}

export function updateTab(tab) {

}

export function onCreateDB() {
    const modalValue = {
        visibleModal: workbenchAction.OPEN_CREATE_DATABASE,
        modalData: null,
    }
    return updateModal(modalValue);
}

export function onCreateDataMap() {

}

export function onCreateTable(params) {
    return (dispatch, getStore) => {
        const {workbench} = getStore();
        const { tabs } = workbench.mainBench;

        let createTableTabIndex = 0;
        for(let i = 0;i<tabs.length;i++){
            if(tabs[i].actionType === workbenchAction.CREATE_TABLE){
                createTableTabIndex = tabs[i].createTableTabIndex > createTableTabIndex?tabs[i].createTableTabIndex:createTableTabIndex
            }
        }

        const newCreateTableTabData = {
            id: moment().unix(),
            name: '新建表',
            createTableTabIndex: createTableTabIndex + 1,
            actionType: workbenchAction.CREATE_TABLE,
        }
        console.log(newCreateTableTabData)

        dispatch(openTab(newCreateTableTabData))
    }
}

export function onSQLQuery() {
    return (dispatch, getStore) => {
        const { workbench } = getStore();
        console.log('onSqlQuery:', workbench);
        const { tabs } = workbench.mainBench;

        let sqlQueryTabIndex = 0;
        for (let i = 0; i < tabs.length; i++) {
            if (tabs[i].actionType === workbenchAction.OPEN_SQL_QUERY) {
                if (tabs[i].sqlQueryTabIndex > sqlQueryTabIndex) {
                    sqlQueryTabIndex = tabs[i].sqlQueryTabIndex;
                }
            }
        }
        const defaultSQLQueryTabData = {
            id: moment().unix(),
            name: `Query ${sqlQueryTabIndex + 1}`,
            sqlQueryTabIndex: sqlQueryTabIndex + 1,
            actionType: workbenchAction.OPEN_SQL_QUERY,
        }

        console.log(defaultSQLQueryTabData)
        dispatch(openTab(defaultSQLQueryTabData));
    }
}


/**
 * 加载左侧树形目录数据
 */
export function loadCatalogue(params) {
    return async (dispatch) => {

        const defaultParams = {
            nodePid: 0,
            level: 2,
        }
        const reqParams = params ? Object.assign(defaultParams, params) : defaultParams;
        const res = await API.loadCatalogue(reqParams);

        if (res.code === 1) {
            dispatch({
                type: workbenchAction.LOAD_CATALOGUE_DATA,
                payload: res.data,
            })
        }
    }
}

/**
 * 保存新建表数据到storage
 */
export function saveNewTableData(params) {
    console.log(params)
    return (dispatch)=>{
        dispatch({
            type: workbenchAction.NEW_TABLE_INFO_CHANGE,
            payload: params
        })
    }
}

export function handleNextStep() {
    console.log('sdsdadsaNEXT')
    return (dispatch,getStore) => {
        const { workbench } = getStore();
        const { currentStep } = workbench.mainBench;
        if(currentStep === 3){
            //提交表单
        }else{
            return dispatch({
                type: workbenchAction.NEXT_STEP
            })
        }
    }
}

export function handleLastStep(){
    return (dispatch,getStore) => {
        const { workbench } = getStore();
        const { currentStep } = workbench.mainBench;

        return dispatch({
            type: workbenchAction.LAST_STEP
        })
    }
}

/**
 * 获取表详情
 */
export function getTableDetail(){
    return async (dispatch,getStore)=>{
        const { workbench } = getStore();
        const res = await API.getTableDetail({
            tableId,
        })
        if (res.code === 1){
            return dispatch({
                type: workbenchAction.GET_TABLE_DETAIL,
                payload: res.data
            })
        }
    }
}



/**
 * 生成建表语句
 */
export function onGenerateCreateSQL(tableId) {

    return async dispatch => {

        const res = await API.getCreateSQL({
            tableId,
        });
    
        if (res.code === 1) {
            const modalValue = {
                visibleModal: workbenchAction.GENERATE_CREATE_SQL,
                modalData: res.data,
            }
            return dispatch(updateModal(modalValue))
        } else {
            notification.error({
                message: '提示',
                description: '生成建表语句失败！'
            });
        }
        return dispatch(resetModal());
    }
}

/**
 * 存储新建表数据至服务端
 */
export function handleSave(){
    return (dispatch,getStore) => {

        const { workbench } = getStore();
        const { newanalyEngineTableDataList, currentTab } = workbench.mainBench;

        const res = API.saveNewTable(newanalyEngineTableDataList[`tableItem${currentTab}`])
        if(res.code === 1){
            return dispatch({
                type: workbenchAction.NEW_TABLE_SAVED
            })
        }
    }
}
/**
 * 编辑表页-保存编辑状态
 * @param {更改的参数对象} params 
 */
export function saveEditTableInfo(params){
    return dispatch => {
        return dispatch({
            type: workbenchAction.SAVE_EDITTABLE_INFO,
            payload: params
        })
    }
}
/**
 * 保存表信息
 * @param {预留} param 
 */
export function saveTableInfo(param){
    return (dispatch,getStore)=>{
        const { workbench } = getStore();
        const { editTableInfoList, currentTab } = workbench.mainBench;
    
        const params = editTableInfoList[`tableInfo${currentTab}`]
        const res = API.saveTableInfo(params);
        
        if(res.code === 1){
            return dispatch({
                type: workbenchAction.TABLE_INFO_MOTIFIED
            })
        }
    }
}