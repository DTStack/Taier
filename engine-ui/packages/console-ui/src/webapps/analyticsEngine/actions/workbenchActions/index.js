import * as comm from './comm';
import * as database from './database';
import * as table from './table';
import * as datamap from './datamap';

export default Object.assign(
    {},
    comm,
    table,
    datamap,
    database,
)