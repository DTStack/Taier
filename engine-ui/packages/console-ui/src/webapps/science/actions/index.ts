// import commonActionType from '../consts/commonActionType';
// import API from '../api';

const commonActions: any = {
    getUserList (params: any) {
        return (dispatch: any) => {
            // API.getUserList(params).then((res: any) => {
            //     if (res.code === 1) {
            //         dispatch({
            //             type: commonActionType.GET_USER_LIST,
            //             payload: res.data
            //         });
            //     }
            // });
        };
    },
    getAllDict (params: any) {
        return (dispatch: any) => {
            // API.getAllDict(params).then((res: any) => {
            //     if (res.code === 1) {
            //         dispatch({
            //             type: commonActionType.GET_ALL_DICT,
            //             payload: res.data
            //         });
            //     }
            // });
        };
    }
};

export default commonActions;
