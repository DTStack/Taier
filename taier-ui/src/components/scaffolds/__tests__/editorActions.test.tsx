import { render } from '@testing-library/react';
import { omit } from 'lodash';
import actions from '../editorActions';

describe("Test editor's actions", () => {
    Object.keys(actions).forEach((action) => {
        const component = actions[action];

        it(`Match ${component.id} snapshot`, () => {
            if (typeof component.icon === 'string') {
                expect(component).toMatchSnapshot();
            } else {
                expect(
                    render(
                        <>
                            <pre>{JSON.stringify(omit(component, ['icon']))}</pre>
                            {component.icon}
                        </>
                    ).asFragment()
                ).toMatchSnapshot();
            }
        });
    });
});
