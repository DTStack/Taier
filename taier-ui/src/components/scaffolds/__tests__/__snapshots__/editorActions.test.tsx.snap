// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test editor's actions Match task.convert.script snapshot 1`] = `
<DocumentFragment>
  <pre>
    {"id":"task.convert.script","name":"转换为脚本","title":"转换为脚本","place":"outer"}
  </pre>
  <span
    aria-label="swap"
    class="anticon anticon-swap"
    role="img"
  >
    <svg
      aria-hidden="true"
      data-icon="swap"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M847.9 592H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h605.2L612.9 851c-4.1 5.2-.4 13 6.3 13h72.5c4.9 0 9.5-2.2 12.6-6.1l168.8-214.1c16.5-21 1.6-51.8-25.2-51.8zM872 356H266.8l144.3-183c4.1-5.2.4-13-6.3-13h-72.5c-4.9 0-9.5 2.2-12.6 6.1L150.9 380.2c-16.5 21-1.6 51.8 25.1 51.8h696c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`Test editor's actions Match task.format snapshot 1`] = `
<DocumentFragment>
  <pre>
    {"id":"task.format","name":"格式化","title":"格式化","place":"outer"}
  </pre>
  <span
    aria-label="format-painter"
    class="anticon anticon-format-painter"
    role="img"
  >
    <svg
      aria-hidden="true"
      data-icon="format-painter"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <defs>
        <style />
      </defs>
      <path
        d="M840 192h-56v-72c0-13.3-10.7-24-24-24H168c-13.3 0-24 10.7-24 24v272c0 13.3 10.7 24 24 24h592c13.3 0 24-10.7 24-24V256h32v200H465c-22.1 0-40 17.9-40 40v136h-44c-4.4 0-8 3.6-8 8v228c0 .6.1 1.3.2 1.9A83.99 83.99 0 00457 960c46.4 0 84-37.6 84-84 0-2.1-.1-4.1-.2-6.1.1-.6.2-1.2.2-1.9V640c0-4.4-3.6-8-8-8h-44V520h351c22.1 0 40-17.9 40-40V232c0-22.1-17.9-40-40-40zM720 352H208V160h512v192zM477 876c0 11-9 20-20 20s-20-9-20-20V696h40v180z"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`Test editor's actions Match task.run snapshot 1`] = `
Object {
  "disabled": false,
  "icon": "play",
  "id": "task.run",
  "name": "Run Task",
  "place": "outer",
  "title": "运行",
}
`;

exports[`Test editor's actions Match task.run snapshot 2`] = `
Object {
  "disabled": true,
  "icon": "loading~spin",
  "id": "task.run",
  "name": "Running",
  "title": "运行中",
}
`;

exports[`Test editor's actions Match task.save snapshot 1`] = `
Object {
  "icon": "save",
  "id": "task.save",
  "name": "Save Task",
  "place": "outer",
  "title": "保存",
}
`;

exports[`Test editor's actions Match task.stop snapshot 1`] = `
Object {
  "disabled": true,
  "icon": "debug-pause",
  "id": "task.stop",
  "name": "Stop Task",
  "place": "outer",
  "title": "停止运行",
}
`;

exports[`Test editor's actions Match task.submit snapshot 1`] = `
<DocumentFragment>
  <pre>
    {"id":"task.submit","name":"提交至调度","title":"提交至调度","place":"outer"}
  </pre>
  <span
    aria-label="upload"
    class="anticon anticon-upload"
    role="img"
  >
    <svg
      aria-hidden="true"
      data-icon="upload"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M400 317.7h73.9V656c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V317.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 163a8 8 0 00-12.6 0l-112 141.7c-4.1 5.3-.4 13 6.3 13zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`Test editor's actions Match task.syntax snapshot 1`] = `
<DocumentFragment>
  <pre>
    {"id":"task.syntax","name":"语法检查","title":"语法检查","place":"outer"}
  </pre>
  <span
    style="font-size: 0px;"
    title="语法检查"
  >
    <svg
      class="icon"
      height="1em"
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M259.413 677.376l46.08-126.976h212.224l46.08 127.019h88.747L458.837 170.667h-94.464L170.667 677.376h88.746zm233.472-195.157h-162.56l80.214-222.123h2.858l79.488 222.123z"
        fill="currentColor"
      />
      <path
        d="M831.019 588.203a32 32 0 0145.226 45.013l-3.328 3.413L578.22 891.563a32 32 0 01-41.43.384l-3.37-3.286-105.515-118.016a32 32 0 0144.288-45.952l3.413 3.286 84.523 94.549 270.933-234.325z"
        fill="currentColor"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`Test editor's actions Match task_ops snapshot 1`] = `
<DocumentFragment>
  <pre>
    {"id":"task_ops","name":"运维","title":"运维","place":"outer"}
  </pre>
  <span
    aria-label="login"
    class="anticon anticon-login"
    role="img"
  >
    <svg
      aria-hidden="true"
      data-icon="login"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <defs>
        <style />
      </defs>
      <path
        d="M521.7 82c-152.5-.4-286.7 78.5-363.4 197.7-3.4 5.3.4 12.3 6.7 12.3h70.3c4.8 0 9.3-2.1 12.3-5.8 7-8.5 14.5-16.7 22.4-24.5 32.6-32.5 70.5-58.1 112.7-75.9 43.6-18.4 90-27.8 137.9-27.8 47.9 0 94.3 9.3 137.9 27.8 42.2 17.8 80.1 43.4 112.7 75.9 32.6 32.5 58.1 70.4 76 112.5C865.7 417.8 875 464.1 875 512c0 47.9-9.4 94.2-27.8 137.8-17.8 42.1-43.4 80-76 112.5s-70.5 58.1-112.7 75.9A352.8 352.8 0 01520.6 866c-47.9 0-94.3-9.4-137.9-27.8A353.84 353.84 0 01270 762.3c-7.9-7.9-15.3-16.1-22.4-24.5-3-3.7-7.6-5.8-12.3-5.8H165c-6.3 0-10.2 7-6.7 12.3C234.9 863.2 368.5 942 520.6 942c236.2 0 428-190.1 430.4-425.6C953.4 277.1 761.3 82.6 521.7 82zM395.02 624v-76h-314c-4.4 0-8-3.6-8-8v-56c0-4.4 3.6-8 8-8h314v-76c0-6.7 7.8-10.5 13-6.3l141.9 112a8 8 0 010 12.6l-141.9 112c-5.2 4.1-13 .4-13-6.3z"
      />
    </svg>
  </span>
</DocumentFragment>
`;
